source/DSCResources/DSC_CMAssetIntelligencePoint/DSC_CMAssetIntelligencePoint.psm1

```diff
@@ -0,0 +1,377 @@
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteServerName
+    )
+
+    Write-Verbose -Message $script:localizedData.RetrieveSettingValue
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    $ap = Get-CMAssetIntelligenceSynchronizationPoint -SiteCode $SiteCode -SiteSystemServerName $SiteServerName
+
+    if ($ap)
+    {
+        $apProps      = Get-CMAssetIntelligenceProxy | Where-Object -FilterScript {$_.ProxyName -eq $SiteServerName}
+        $cert         = $apProps.ProxyCertPath
+        $apEnabled    = $apProps.ProxyEnabled
+        $syncEnabled  = $apProps.PeriodicCatalogUpdateEnabled
+        $syncSchedule = $apProps.PeriodicCatalogUpdateSchedule
+        $status       = 'Present'
+
+        $convertCimParam = @{
+            ScheduleString = $syncSchedule
+            CimClassName   = 'DSC_CMAssetIntelligenceSynchronizationSchedule'
+        }
+        $sSchedule = ConvertTo-CimCMScheduleString @convertCimParam
+    }
+
+    else
+    {
+        $status = 'Absent'
+    }
+
+    return @{
+        SiteServerName        = $SiteServerName
+        SiteCode              = $SiteCode
+        CertificateFile       = $cert
+        Enable                = $apEnabled
+        EnableSynchronization = $syncEnabled
+        Schedule              = $sSchedule
+        Ensure                = $status
+    }
+}
+
+ <#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies a site code for the Configuration Manager site that manages the system role for the asset intelligence point.
+
+    .PARAMETER SiteServerName
+        Specifies the Site Server to install or configure the role on.
+
+    .PARAMETER CertificateFile
+        Specifies the path to a System Center Online authentication certificate (.pfx) file.
+        If used, this must be in UNC format. Local paths are not allowed.
+
+    .PARAMETER Schedule
+        Specifies when the asset intelligence catalog is synchronized.
+
+    .PARAMETER Enable
+        Specifies whether the installed asset intelligence role is enabled or disabled.
+
+    .PARAMETER EnableSynchronization
+        Specifies whether to synchronize the asset intelligence catalog.
+
+    .PARAMETER Ensure
+        Specifies whether the asset intelligence synchronization point is present or absent.
+
+    .Notes
+        This role must only be installed on top-level site of the hierarchy
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteServerName,
+
+        [Parameter()]
+        [String]
+        $CertificateFile,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $Schedule,
```

Comment:
what if I want to specify None for pulling schedule?

Reply:
Added changes to account for nonrecurring schedules